pkg/context/detached_test.go

```diff
@@ -0,0 +1,61 @@
+//go:build test_unit
+
+package context
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/nuclio/logger"
+	nucliozap "github.com/nuclio/zap"
+	"github.com/stretchr/testify/suite"
+)
+
+type DetachedTestSuite struct {
+	suite.Suite
+	logger logger.Logger
+}
+
+func (suite *DetachedTestSuite) SetupTest() {
+	suite.logger, _ = nucliozap.NewNuclioZapTest("test")
+}
+
+func (suite *DetachedTestSuite) TestDetachedOnCancelledParent() {
+	timerChannel := make(chan int)
+	expectedWaitTime := 5
+
+	parentCtx, cancelParentCtx := context.WithCancel(context.Background())
+	childCtx, cancelChildCtx := context.WithCancel(NewDetached(parentCtx))
+
+	go suite.measureTimeUntilCancellation(childCtx, timerChannel)
+
+	suite.logger.Debug("Cancelling parent context")
+	cancelParentCtx()
+	time.Sleep(6 * time.Second)
+	suite.logger.Debug("Cancelling child context")
+	cancelChildCtx()
+
+	suite.Require().Equal(expectedWaitTime, <-timerChannel)
```

Comment:
well, this part of the test might wait a little bit longer on cpu and the child would tick some more times.
what you can do, is cancel parent and check if `child.Done()` - simple.
then cancel child and make sure `child.Done` has something
no need for a timer in the child routine